ts-node-plugin/index.js

```diff
@@ -0,0 +1,38 @@
+const {transform} = require("../dist");
+
+const ModuleKindCommonJS = 1;
+
+/**
+ * ts-node transpiler plugin
+ */
+function create(createOptions) {
+  const {nodeModuleEmitKind} = createOptions;
+  const {module} = createOptions.service.config.options;
+  return {
+    transpile(input, transpileOptions) {
+      const {fileName} = transpileOptions;
+      const transforms = ["typescript"];
+      if (module === ModuleKindCommonJS || nodeModuleEmitKind === "nodecjs") {
+        transforms.push("imports");
+      }
+      if (fileName.endsWith(".tsx") || fileName.endsWith(".jsx")) {
+        transforms.push("jsx");
```

Comment:
Currently, Sucrase only supports the old JSX transform; see #585. If the tsconfig is configured for the new transform, an open question is whether to crash with a message that it's unsupported, or if it's better to just transpile with the old transform anyway. Anyone using the new transform likely has at least one file without `import React from "react";` (or is using `jsxImportSource`), so it's probably better to just crash here to give a more helpful error message.

Reply:
A related thing I want to get right: ensuring plugins have a good way to raise diagnostics and / or warnings.  ts-node doesn't do many warnings today, but I want to add a good mechanism for it.  So that ts-node and/or transpiler plugins can warn users about potential config issues, version incompatibilities, stuff like that.  And then we can support config flags to suppress warnings either completely, or suppress warnings by diagnostic code.

What this would mean for sucrase: you have the *option* of returning either errors or warnings from either `create()` or `transpile()` and ts-node will handle either throwing or logging, or suppressing based on tsconfig.  I'll try to jot down ideas in https://github.com/TypeStrong/ts-node/issues/1851